src/connectors/eggs.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+Connector.getDuration = () => document.querySelector('#aPlayer').duration;
+
+Connector.getCurrentTime = () => document.querySelector('#aPlayer').currentTime;
+
+Connector.playerSelector = '.m-upload-list';
+
+if (window.location.href.includes('/artist/')) {
+	setupArtistPlayer();
+} else {
+	setupSongPlayer();
+}
+
+function setupArtistPlayer() {
+
+	Connector.getTrackArt = () => document.querySelector('.pause[style*="display: block;"]').closest('.img-album').querySelector('img').src;
```

Comment:
This could be improved by overriding `Connector.getTrackInfo` method:
```
Connector.getTrackInfo = () => {
	const songItem = /* get song item */;

	/* get all properties from songItem here */

	return { artist, track, trackArt };
};
```

Also, I would get the parent element by selecting `[id^=song] .pause[style*="display: block;"]` to avoid using `closest` method. But I'm not sure if all song items will match this selector.

Reply:
> Also, I would get the parent element by selecting [id^=song] .pause[style*="display: block;"] to avoid using closest method. But I'm not sure if all song items will match this selector.

I mean i could move the selectors to id^=song, but I would still have to use closest no? As far as I know there is no way to get the parent of a matched element in css (well, other than :has()). `[id^=song] .pause[style*="display: block;"]` will match the pause button itself, not the li element. Unless i'm missing something.

As for the other suggestions, on it.

Reply:
> [id^=song] .pause[style*="display: block;"] will match the pause button itself, not the li element.

Yes, you're right. Using the `closest` method is fine in this case.